### PR TITLE
1460 Course enrichment factory issues

### PR DESCRIPTION
**Pull request by davidgisbey**
### Context

Pull request no. 5 to clean up the associations in our factories. This PR removes with_course_enrichments from the course factory.

### Changes proposed in this pull request

- Remove with_course_enrichments from the course factory

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by defong** [comment]
[[M] Improve how we create factories & related data in specs.](https://trello.com/c/PhV37qzp/1460-m-improve-how-we-create-factories-related-data-in-specs)

File: spec/models/course/publish_enrichments_spec.rb [hunk 1]
```diff
@@ -25,27 +25,32 @@
...
     end
 
     context "for a course an initial draft enrichments" do
-      subject { create(:course, with_enrichments: [[:initial_draft]]) }
+      let(:course) { create(:course, enrichments: [build(:course_enrichment, :initial_draft)]) }
+      subject { course }
+
+      it "set to draft" do
+        expect(subject.content_status).to eq(:draft)
+      end
```
Comment (misaka) [inline spec/models/course/publish_enrichments_spec.rb]: Duplication of the test below?

File: spec/models/course/publish_enrichments_spec.rb [hunk 2]
```diff
@@ -2,17 +2,17 @@
...
-               [:subsequent_draft, created_at: 1.day.ago],
-             ]).enrichments
-    }
+    let(:first_enrichment) { build(:course_enrichment, :published, created_at: 5.days.ago) }
+    let(:second_enrichment) { build(:course_enrichment, :published, created_at: 3.days.ago) }
+    let(:third_enrichment) { build(:course_enrichment, :subsequent_draft, created_at: 1.day.ago) }
+
+    let(:course) { create(:course, enrichments: [first_enrichment, second_enrichment, third_enrichment]) }
+
+    subject { course.reload.enrichments }
```
Comment (misaka) [inline spec/models/course/publish_enrichments_spec.rb]: Looking at the changes to this file in general, I probably would have done something like this:

```
describe Course, type: :model do
  let(:enrichments { [] }
  let(:course)     { create(:course, enrichments: enrichments) }

  subject { course }

  describe "#enrichments" do
    let(:enrichments) do
      [
        build(:course_enrichment, :published, created_at: 5.days.ago)
        build(:course_enrichment, :published, created_at: 3.days.ago)
        build(:course_enrichment, :subsequent_draft, created_at: 1.day.ago)
      ]
    end

    ...
  end

  ...

  context "for a course an initial draft enrichments" do
    let(:enrichments) { [build(:course_enrichment, :initial_draft)]) }
    ...
  end

  ...
```

The idea being that we create the course the same way, and the subject is the same, it's just the `enrichments` that change in the context blocks. Not a blocker though if you want to see the back of these tests.

File: spec/models/course/publish_enrichments_spec.rb (hunk 1 above)
Comment (defong) [reply spec/models/course/publish_enrichments_spec.rb]: opps

File: spec/models/course/publish_enrichments_spec.rb (hunk 2 above)
Comment (defong) [reply spec/models/course/publish_enrichments_spec.rb]: slapped in enrichments

File: spec/models/course/publish_enrichments_spec.rb [hunk 3]
```diff
@@ -25,27 +26,40 @@
...
-      subject { create(:course, with_enrichments: []) }
+      subject { create(:course) }
       its(:content_status) { should eq(:empty) }
     end
 
     context "for a course an initial draft enrichments" do
-      subject { create(:course, with_enrichments: [[:initial_draft]]) }
+      let(:enrichments) { [build(:course_enrichment, :initial_draft)] }
+      let(:course) { create(:course, enrichments: enrichments) }
+      subject { course }
```
Comment (misaka) [inline spec/models/course/publish_enrichments_spec.rb]: This is now the same as what's on line 10-12 so can probably be removed.

Comment (defong) [reply spec/models/course/publish_enrichments_spec.rb]: not quite, but yeah

**Pull request merged**